SimG4Core/Notification/interface/TrackWithHistory.h

```diff
@@ -57,11 +60,20 @@ class TrackWithHistory {
   const math::XYZTLorentzVectorF &getMomentumAtBoundary() const { return momentumAtBoundary_; }
   int getIDAtBoundary() const { return idAtBoundary_; }
 
+  // tracker surface
+  const math::XYZVectorD &trackerSurfacePosition() const { return tkSurfacePosition_; }
```

Comment:
@civanch where do you intend to use it? I fail to see its use in this PR

Reply:
@fabiocos , this surface data are a part of SimTrack - persistent MC truth. Until now these was not filled at all and unused. In the current PR I have added the same variables to TrackWithHistory in order to allow Phase2 detectors to use this for a good purpose. The name of course misleading, may be we would change SimTrack somehow adding information needed both for MTD and HGCal.

Reply:
In the next iteration I want to drop TmpSimTrack and use TrackWithHistory to fill SimTrack.